runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/widgets/SuggestProcessesNameWidget.tsx

```diff
@@ -1,54 +1,68 @@
-import React, { FC, useMemo, useEffect } from 'react';
+import React, { FC, useMemo, useEffect, useRef, useState } from 'react';
 
 import { TextField, Autocomplete } from '@mui/material';
 import { WidgetProps } from '@rjsf/core';
 
-import { IFormData } from '#src-app/Actions/types';
+import { BotSystem } from 'runbotics-common';
+
+import useTranslations from '#src-app/hooks/useTranslations';
 import { useDispatch, useSelector } from '#src-app/store';
 import { processActions } from '#src-app/store/slices/Process';
 
-interface CustomWidgetProps extends WidgetProps {
-    formData: IFormData;
-}
-
-const ProcessNameSuggestionWidget: FC<CustomWidgetProps> = (props) => {
+const ProcessNameSuggestionWidget: FC<WidgetProps> = (props) => {
     const dispatch = useDispatch();
     const { byId: processes } = useSelector((state) => state.process.all);
+    const [customError, setCustomError] = useState('');
+    const [value, setValue] = useState(props.value);
+    const { translate } = useTranslations();
 
     useEffect(() => {
         dispatch(processActions.getProcesses());
     // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
     const {
-        process: { id: processId },
+        process: { id: processId, system: { name: rootProcessSystem } },
     } = useSelector((state) => state.process.draft);
 
+    const isProcessSystemCompatible = (processSystem: string): boolean => (
+        rootProcessSystem === processSystem ||
+        rootProcessSystem === BotSystem.ANY ||
```

Comment:
now if parent process is any, and child process is linux and bot is windows there'll be problem again

rootProcessSystem === BotSystem.ANY -> this has to be removed.

Reply:
If I remove this condition, in case parent process  is ANY, the options will show only those processes which are ANY, so this is also not good. I assumed that if some1 select parent process ANY and child process is LINUX but bot system is still WINDOWS, scheduler will return proper exception and inform user in UI.

Reply:
good, that there's proper exception and information for user.

but this shows a weak point - I'll get back to u